src/sync.rs

```diff
@@ -42,6 +43,14 @@ where
     IO: DfuIo<Read = usize, Write = usize, Reset = (), Error = E>,
     E: From<std::io::Error> + From<Error>,
 {
+    /// Download a slice to on to the device
+    pub fn write_flash_from_slice(&mut self, slice: &[u8]) -> Result<(), IO::Error> {
+        let length = slice.len();
+        let cursor = Cursor::new(slice);
+
+        self.download(cursor, length as u32)
```

Comment:
We should probably use `u32::try_from` here and convert the error to an IO error of kind `InvalidData`: https://doc.rust-lang.org/std/io/enum.ErrorKind.html#variant.InvalidData

If we don't do that, files that exceed the limit will make the program panic (which is not nice).